app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -176,3 +184,34 @@ class CiYaml {
     }
   }
 }
+
+/// Class to expose the PinnedVersion method for testing.
+class DependencyValidator {
+  /// Dependency is guaranteed to be non empty as it must be found before this
+  /// method is called.
+  ///
+  /// Checks a dependency string for a pinned version.
+  /// If a version is found then it must not be empty or 'latest.'
+  static void hasVersion({required String dependencyJsonString}) {
+    final List<String> exceptions = <String>[];
+    // decoded will contain a list of maps
+    dynamic decoded = json.decode(dependencyJsonString);
+
+    for (Map<String, dynamic> depMap in decoded) {
+    
+      if (!depMap.containsKey('version')) {
+        exceptions.add('ERROR: dependency ${depMap['dependency']} must have a version.');
+      } else {
+        String version = depMap['version'] as String;
+        if (version.isEmpty || version == 'latest') {
+          exceptions.add('ERROR: dependency ${depMap['dependency']} must have a non empty, non "latest" version supplied.');
+        }
+      }
+    }
+
+    // if (exceptions.isNotEmpty) {
+    //   final String fullException = exceptions.reduce((String exception, _) => exception + '\n');
+    //   throw FormatException(fullException);
+    // }
```

Comment:
Nit: remove.

Reply:
My fault. I mean to keep this. Good catch though.